Add warning log on dropped spans in batch span processor (#3289)
open-telemetry/opentelemetry-go: CHANGELOG.md

## [Unreleased]

### Added

> - The `go.opentelemetry.io/sdk/trace/batch_span_processor` will now keep track of known dropped span count and log during batch export on newly dropped spans. (#3289)

Reviewer: ```suggestion
- The `BatchSpanProcessor` from `go.opentelemetry.io/otel/sdk/trace` logs the spans dropped in a batch every export with verbosity `1`. (#3289)
```

## [Unreleased]

> - Batch span processor will now keep track of known dropped span count and log a warning during batch export on newly dropped spans. (#3289)

Reviewer: Please include this in a sub section: https://keepachangelog.com/en/1.0.0/

   A non-default registerer can be used by passing the `WithRegisterer` option. (#3239)
- Added the `WithAggregationSelector` option to the `go.opentelemetry.io/otel/exporters/prometheus` package to change the default `AggregationSelector` used. (#3341)
- The Prometheus exporter in `go.opentelemetry.io/otel/exporters/prometheus` converts the `Resource` associated with metric exports into a `target_info` metric. (#3285)

> - Batch span processor will now keep track of known dropped span count and log a warning during batch export on newly dropped spans. (#3289)

Reviewer 2: This needs to be moved up into the `Unreleased` section.
